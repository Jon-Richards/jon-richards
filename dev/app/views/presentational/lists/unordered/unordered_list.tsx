/** @jsx jsx */

import * as React from 'react';
import { List } from '../interfaces/list';
import { ListItem } from '../interfaces/list_item';
import { jsx } from '@emotion/core';
import { STYLES } from './styles';

export function UnorderedList(props: UnorderedListProps): JSX.Element {
  const items = mapItems(props.items);
  return (
    <ul title={props.title} data-testid="unordered-list" css={STYLES.list}>
      {items}
    </ul>
  )
}

export type UnorderedListProps = List<ListItem>;

function mapItems(items: ListItem[]): JSX.Element {
  const result = items.map(item => (
    <li key={item.id} data-testid="unordered-list__item" css={STYLES.item}>
      {item.content}
    </li>
  ));
  return <React.Fragment>{result}</React.Fragment>
}
