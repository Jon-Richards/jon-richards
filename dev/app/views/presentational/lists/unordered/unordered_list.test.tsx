import * as React from 'react';
import { UnorderedList } from './unordered_list';
import { render } from 'enzyme';

describe('The OrderedList component.', () => {
  it('Should match the snapshot.', () => {
    const wrapper = render(
      <UnorderedList
        title="An Ordered List."
        items={[
          { id: 'one', content: 'Item one' },
          { id: 'two', content: 'Item one' }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render one list item for each element in the items array.', () => {
    const wrapper = render(
      <UnorderedList
        title="An Ordered List."
        items={[
          { id: 'one', content: 'Item one' },
          { id: 'two', content: 'Item one' }
        ]}
      />
    );
    const testHook = '[data-testid="unordered-list__item"]';

    expect(wrapper.find(testHook).length).toBe(2);
  });
});
