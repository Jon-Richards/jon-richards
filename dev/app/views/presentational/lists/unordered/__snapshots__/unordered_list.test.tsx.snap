// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The OrderedList component. Should match the snapshot. 1`] = `
.emotion-2 {
  list-style-type: circle;
}

.emotion-0 {
  display: list-item;
  font-family: "Open Sans",sans-serif;
  font-weight: 400;
  line-height: 1.6rem;
  padding-left: 0.5rem;
  margin-left: 2rem;
  margin-bottom: calc(2rem / 2);
}

.emotion-0::before {
  right: 1rem;
  opacity: 0.5;
}

<ul
  class="emotion-2"
  data-testid="unordered-list"
  title="An Ordered List."
>
  <li
    class="emotion-0"
    data-testid="unordered-list__item"
  >
    Item one
  </li>
  <li
    class="emotion-0"
    data-testid="unordered-list__item"
  >
    Item one
  </li>
</ul>
`;
